packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_model.dart

```diff
@@ -3,37 +3,180 @@
+
+class LinkData with SearchableDataMixin {
   LinkData({
-    required this.os,
     required this.domain,
-    required this.paths,
-    this.scheme = 'Http://, Https://',
+    required this.path,
+    required this.os,
+    this.scheme = const <String>['Http://', 'Https://'],
     this.domainError = false,
     this.pathError = false,
   });
 
-  final String os;
-  final List<String> paths;
+  final String path;
   final String domain;
-  final String scheme;
+  final List<String> os;
+  final List<String> scheme;
   final bool domainError;
   final bool pathError;
 
-  String get searchLabel => (os + paths.join() + domain + scheme).toLowerCase();
+  @override
+  bool matchesSearchToken(RegExp regExpSearch) {
+    return (domain.caseInsensitiveContains(regExpSearch) == true) ||
+        (path.caseInsensitiveContains(regExpSearch) == true);
+  }
 
-  LinkData mergeByDomain(LinkData? linkData) {
-    if (linkData == null) {
-      return this;
-    }
+  @override
+  String toString() => 'LinkData($domain $path)';
+}
+
+class DomainColumn extends ColumnData<LinkData> {
+  DomainColumn()
+      : super(
+          'Domain',
+          fixedWidthPx: scaleByFontFactor(kDeeplinkTableCellDefaultWidth),
+        );
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  String getValue(LinkData dataObject) => dataObject.domain;
+
+  @override
+  Widget getDisplayWidget(LinkData dataObject, BuildContext context) {
+    return Row(
+      children: [
+        if (dataObject.domainError)
+          Padding(
+            padding: const EdgeInsets.only(right: denseSpacing),
+            child: Icon(
+              Icons.error,
+              color: Theme.of(context).colorScheme.error,
+            ),
+          ),
+        const SizedBox(width: denseSpacing),
+        Text(
+          dataObject.domain,
+          overflow: TextOverflow.ellipsis,
+        ),
+      ],
+    );
+  }
+}
+
+class PathColumn extends ColumnData<LinkData> {
+  PathColumn()
+      : super(
+          'Path',
+          fixedWidthPx: scaleByFontFactor(kDeeplinkTableCellDefaultWidth),
+        );
+
+  @override
+  bool get supportsSorting => true;
 
-    assert(domain == linkData.domain);
+  @override
+  String getValue(LinkData dataObject) => dataObject.path;
 
-    return LinkData(
-      os: os,
-      domain: domain,
-      paths: [...paths, ...linkData.paths],
-      domainError: domainError || linkData.domainError,
+  @override
+  Widget getDisplayWidget(LinkData dataObject, BuildContext context) {
+    return Row(
+      children: [
+        if (dataObject.pathError)
+          Padding(
+            padding: const EdgeInsets.only(right: denseSpacing),
+            child: Icon(
+              Icons.error,
+              color: Theme.of(context).colorScheme.error,
+            ),
+          ),
+        const SizedBox(width: denseSpacing),
+        Text(
+          dataObject.path,
+          overflow: TextOverflow.ellipsis,
+        ),
+      ],
```

Comment:
This is repeated above. Please create a helper widget `_ErrorAwareText` or similar that you can use in both DomainColumn and PathColumn